### PR TITLE
DOSBox-X: Add Package

### PR DESCRIPTION
**Pull request by OCsonic**

### CODE REVIEW

**Comment by OCsonic** [comment]
for package request https://github.com/termux/termux-packages/pull/10875

File: x11-packages/dosbox-x/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+TERMUX_PKG_HOMEPAGE=https://dosbox-x.com/
+TERMUX_PKG_DESCRIPTION="DOSBox-X fork of the DOSBox project"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.83.22
+TERMUX_PKG_REVISION=1
```
Comment (MrAdityaAlok) [inline x11-packages/dosbox-x/build.sh:6]: Remove this. This is only needed for changes made after package is published (for `dpkg` to consider it as update).

File: x11-packages/dosbox-x/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+TERMUX_PKG_SRCURL=https://github.com/joncampbell123/dosbox-x/archive/refs/tags/dosbox-x-v${TERMUX_PKG_VERSION/.3/-3}.tar.gz
+TERMUX_PKG_SHA256=
```
Comment (MrAdityaAlok) [inline x11-packages/dosbox-x/build.sh:8]: This is necessary. CI will fail without it.

File: x11-packages/dosbox-x/build.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+TERMUX_PKG_DEPENDS="libc++, libpng, libiconv, libx11, sdl2, sdl2-net, zlib"
+
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+--disable-avcodec
+--enable-sdl2
+--disable-dynamic-x86
+--disable-fpu-x86
+--disable-opengl
+--prefix=/usr "$@"
```
Comment (MrAdityaAlok) [inline x11-packages/dosbox-x/build.sh:17]: Remove `--prefix=/usr`. It will be passed by configure step.

Also, there is no need of `$@`.

**Review by MrAdityaAlok — CHANGES_REQUESTED** [review]

File: x11-packages/dosbox-x/build.sh (hunk 2 above)
Comment (OCsonic) [reply x11-packages/dosbox-x/build.sh:8]: oops i forgot to add back in the checksum! on it!

File: x11-packages/dosbox-x/build.sh (hunk 3 above)
Comment (MrAdityaAlok) [reply x11-packages/dosbox-x/build.sh:17]: You should read [Creating-new-package](https://github.com/termux/termux-packages/wiki/Creating-new-package) before making PR.

**Comment by OCsonic** [comment]
2 questions:
how do I set -liconv in LDFLAGS with this system?
and how do I download sse2neon.h straight to include/ without redistributing the entire file as a patch?

**Comment by Grimler91** [comment]
> how do I set -liconv in LDFLAGS with this system? 

Should normally be added automatically by the configure in the package script

> how do I download sse2neon.h straight to include/ without redistributing the entire file as a patch?

[sse2neon](https://github.com/DLTcollab/sse2neon) should be added as a separate package, and added to the build-dependencies of dosbox-x

**Comment by suhan-paradkar** [comment]
> 2 questions: how do I set -liconv in LDFLAGS with this system? and how do I download sse2neon.h straight to include/ without redistributing the entire file as a patch?

You can see how `libandoid-glob` does it [here](https://github.com/termux/termux-packages/blob/b5c8832ca4d9e9b70d9d79e0eb89343659e833fb/packages/libandroid-glob/build.sh#L19)

**Comment by suhan-paradkar** [comment]
> [sse2neon](https://github.com/DLTcollab/sse2neon) should be added as a separate package, and added to the build-dependencies of dosbox-x

@OCsonic You need to create the sse2neon package somewhat like in `libandroid-glob` or `libandroid-sysv-semaphore`

**Comment by wmcbtech30** [comment]
You can simply add `LDFLAGS+=" -liconv` at function `termux_step_pre_configure`

And for including `sse2neon.h` from the internet, you can add a function `termux_step_post_get_sources` and use `termux_download <sse2neon.h url> $TERMUX_PREFIX/include/sse2neon.h <sse2neon.h sha256sum>` (replace placeholders with actual download link and sha256sum)

**Comment by OCsonic** [comment]
> > how do I download sse2neon.h straight to include/ without redistributing the entire file as a patch?
> 
> [sse2neon](https://github.com/DLTcollab/sse2neon) should be added as a separate package, and added to the build-dependencies of dosbox-x

I gave it a try and I'm not experienced enough with libraries to know what I'm looking at. Going to need someone else to package the library.

**Comment by Grimler91** [comment]
Sse2neon has issues with i686, so i686 [needs to be blacklisted](https://github.com/termux/termux-packages/commit/0e00ea5ded0ca37967b765f1b917eca85c2ebd11) for dosbox-x as well

**Comment by Grimler91** [comment]
And please [rebase](https://stackoverflow.com/questions/7929369/how-to-rebase-local-branch-onto-remote-master) (to get rid of merge commits) and squash/re-write the commit messages, `removed unnessecary step` and  `remove unnessecary line)` are not very informative (see [git history](https://github.com/termux/termux-packages/commits/master) for formats of previous commits)

File: x11-packages/dosbox-x/build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+TERMUX_PKG_HOMEPAGE=https://dosbox-x.com/
+TERMUX_PKG_DESCRIPTION="DOSBox-X fork of the DOSBox project"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.84.0
+TERMUX_PKG_SRCURL=https://github.com/joncampbell123/dosbox-x/archive/refs/tags/dosbox-x-v${TERMUX_PKG_VERSION/.3/-3}.tar.gz
```
Comment (Grimler91) [inline x11-packages/dosbox-x/build.sh:6]: `${TERMUX_PKG_VERSION/.3/-3}`, /.3/-3 doesn't seem to do anything, please remove it

**Comment by OCsonic** [comment]
I'm working on rebasing and cleaning up commits right now.

**Comment by 2096779623** [comment]
1. Please refer to this to make a patch :

https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#submitting-pull-requests

> 6. Patches: format
> 
> Patches are standard diff output generated by GNU diff or Git. Please avoid editing patches by hand, especially if you don't > > understand format internals.
> 
> Patch is usually created by
> `diff -uNr sourcedir sourcedir.mod > filename.patch`

  
2. Please squash into one commit : https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/squashing-commits  
3. keep the branch up to date(It is recommended to do this before merging this pull requests) : https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch  

image

**Comment by leapofazzam123** [comment]
God, this is painful to look at

**Comment by leapofazzam123** [comment]
Maybe hand it over to @xtkoba instead?

**Comment by leapofazzam123** [comment]
I once successfully build DOSBox-X on Termux but never got to add the package to x11-packages. Maybe just hand it over to me instead?

**Comment by 2096779623** [comment]
> I once successfully build DOSBox-X on Termux but never got to add the package to x11-packages. Maybe just hand it over to me instead?

You can make another PR.

**Comment by OCsonic** [comment]
Sorry for vanishing for over a month, My pc has been broken for awhile but I got my Raspberry PI working again (microSD slot was detached) so I'm working on it from there now. Hopefully will update shortly with the commits squashed and the build script working.

**Comment by OCsonic** [comment]
Can someone reopen the pr, my phone was being crap and it browser github closed it.

**Comment by OCsonic** [comment]
just finished resetting my fork and adding everything as one commit, working on fixing build issues right now

**Comment by OCsonic** [comment]
eh I'll just make a new one since this one is a mess now

**Pull request closed without merging**